git_machete/client.py

```diff
@@ -1698,19 +1700,11 @@ def checkout_github_pr(self, pr_no: int) -> None:
         self.__git.checkout(pr.head)
         print(fmt(f"Pull request `#{pr.number}` checked out at local branch `{pr.head}`"))
 
-    def __get_path_from_pr_chain(self, last_branch: str, all_prs: List[GitHubPullRequest]) -> List[str]:
-        def get_parent_branch_from_pr(branch: str) -> Optional[str]:
-            for pr in all_prs:
-                if pr.head == branch:
-                    return pr.base
-            return None
-
-        path: List[str] = []
-        current_branch: str = last_branch
-        while current_branch:
-            path.append(current_branch)
-            current_branch = get_parent_branch_from_pr(current_branch)
-
+    def __get_path_from_pr_chain(self, current_pr: GitHubPullRequest, all_prs: List[GitHubPullRequest]) -> List[str]:
+        path: List[str] = [current_pr.head]
+        while current_pr:
+            path.append(current_pr.base)
+            current_pr = next(filter(lambda x: x.head == current_pr.base, all_prs), None)
```

Comment:
I think this can be extracted to `utils` next to the things like `flat_map`, under the name `find_or_none`

Reply:
Yes, but then i would need to annotate this function arguments with `GithubPullRequests` and this will cause circular import error.

Reply:
Nope! pls make this function generic, see the mentioned `flat_map`